lib/async_hooks.js

```diff
@@ -366,26 +353,24 @@ function emitBeforeS(asyncId, triggerId = asyncId) {
 
   pushAsyncIds(asyncId, triggerId);
 
-  if (async_hook_fields[kBefore] === 0) {
-    return;
-  }
-
   emitBeforeN(asyncId);
 }
 
 
 // Called from native. The asyncId stack handling is taken care of there before
 // this is called.
 function emitAfterN(asyncId) {
-  if (async_hook_fields[kAfter] > 0) {
-    processing_hook = true;
+  processing_hook = true;
```

Comment:
I’ve added it to `emitAfterS` now; The idea is that the native code doesn’t bother calling `emitAfterN` if the field is 0, so checking it here is pointless for the common case of a call from C++.